### PR TITLE
Fix placements of media query with grid-template-area

### PR DESCRIPTION
**Pull request by YozhikM**
#1048 
cc @ai

### CODE REVIEW

File: lib/hacks/grid-utils.js [hunk 1]
```diff
@@ -329,8 +341,23 @@ function insertAreas(areas, decl, result) {
                     })
                 ));
 
+
```
Comment (ai) [inline lib/hacks/grid-utils.js]: double new line

**Comment by YozhikM** [comment]
@ai I want to sleep, I pressed it wrong :)

**Comment by ai** [comment]
@YozhikM crazy fast :D 

@yepninja need your review here

@Dan503 seems like we have a solution for that the last critical issue

**Comment by Dan503** [comment]
Excellent :)

Once it's released I'll test it and update the article. Thanks :)

File: test/cases/grid-template.css [hunk 2]
```diff
@@ -69,10 +69,26 @@
...
-            [footer-left] "foot" 30px [footer-right]
+            [header-left] 'head' 30px [header-right]
+            [nav-left] 'nav' auto [nav-right]
+            [main-left] 'main' 1fr [main-right]
+            [footer-left] 'foot' 30px [footer-right]
             / 1fr;
     }
 }
+
+.foo {
```
Comment (yepninja) [inline test/cases/grid-template.css]: I think it's better to change existing tests than adding new after all. We already have a large number of tests. And the further the more difficult it becomes to understand them.

Comment (ai) [reply test/cases/grid-template.css]: I like more tests 😄

**Comment by ai** [comment]
Let's decide about the `lib/` code. I will revert quotes in tests. We need this fix ASAP.

File: test/cases/grid-template.css (hunk 2 above)
Comment (yepninja) [reply test/cases/grid-template.css]: And I like more test cases :) In this situation, we duplicate the same test cases many times. We don't increase coverage, just increase the amount of code.

Comment (ai) [reply test/cases/grid-template.css]: Good point 👍

Comment (YozhikM) [reply test/cases/grid-template.css]: I suggest leaving only `.az` and `.aa`

File: test/cases/grid-template-areas.css [hunk 3]
```diff
@@ -24,6 +24,10 @@
 
 @media (max-width: 1000px) {
     .a {
-        grid-template-areas: "head main";
+        grid-template-areas: "head main article";
```
Comment (YozhikM) [inline test/cases/grid-template-areas.css]: I re-used the existed tests

**Comment by ai** [comment]
@YozhikM I was convinced to use single media for all moved areas https://github.com/postcss/autoprefixer/pull/1057

Sorry

**Comment by Dan503** [comment]
What?!? Why?

That's extremely dangerous because it not only moves the grid area code but it also brings all the authors other styles along for the ride. There is a very high risk that this will break the authors styles and then cause resentment amongst the front end community

**Comment by ai** [comment]
@Dan503 single = second

**Comment by ai** [comment]
I mean single for every areas (and one additional for `template-areas`)

**Comment by Dan503** [comment]
Oh right. You were talking about a single duplicate. Not a single media query (ie. Moving the original).

Sorry about the misunderstanding
